lib/msal-common/src/url/UrlString.ts

```diff
@@ -237,6 +237,28 @@ export class UrlString {
         return Constants.EMPTY_STRING;
     }
 
+    /**
+     * Parses query server response string from given string. 
+     * Extract hash between '?code=' and '#' if trailing '# is present.
+     * Returns empty string if no query symbol is found.
+     * @param queryString
+     */
+    static parseQueryServerResponse(queryString: string): string {
+        const queryIndex1 = queryString.indexOf("?code");
```

Comment:
Let's try to refactor this method to be more readable here. I think it can be rewrittent o be more self-documenting and even then it might benefit from comments so anyone who reads it knows why and what it's doing.

Reply:
On further thought: The manual processing of the string may be more readable in general, but we should consider using RegEx. If we do decide to go with RegEx here, we just need to make sure the expression you come up with works for all browsers, since we've had cases where we missed an error with a regular expression in safari.

Reply:
Here's a few thoughts, but I'll preface it by saying: I don't think these differences are super important, your current implementation works well (you can always add a bunch of edge cases to unit tests to test this function) and it would be perfectly fine to leave as is. 

1. I don't think the string size or pattern we're searching for are large enough for performance differences between `indexOf` and RegEx to be a concern here
2. My first thought is "we're trying to match and extract a substring between '?code' or '/?code' and '#'and disregard anything else, which to more of a pattern match (what RegEx does) than a simple check to see if a substring is present in the string (what `indexOf` technically does)
3. A RegEx is a bit tougher to read at a glance (comments that explain what it should do mitigate this), but in this case we have about 10 lines of logic with multiple conditions and index increment operations that can be hard to read and easy to hide a bug in. We'd be trading that for 1 or 2 lines with a RegEx, and there's plenty of tools to debug a RegEx.

Those are just my $0.02, now, assuming you decide against using a RegEx, I also think you can consider a 3rd alternative: leverage the `parseQueryString` method above and check if there's a "code" attribute:


```typescript
const queryString = window.location.url // or wherever we get it from
const serverResponse = UrlString.parseQueryString(queryString);
const authCode = serverResponse.code;

if (authCode) {
// keep going
} else {
// there was no code in the query string, so not an auth code response
}
```

Reply:
parseQueryString returns a string instead of a ServerAuthorizationCodeResponse, so I don't know if we can get a code attribute from that. 
I think I'll keep the current logic, but let me know if there's a way to make it clearer (maybe moving something around or adding some more comments).